recursion/core/src/poseidon2_wide/external.rs

```diff
@@ -381,55 +382,67 @@ where
     AB: SP1RecursionAirBuilder,
 {
     fn eval(&self, builder: &mut AB) {
+        assert!(DEGREE > 3, "Minimum supported constraint degree is 3");
+        let use_sbox_3 = DEGREE < 7;
         let main = builder.main();
         let cols = main.row_slice(0);
-        let cols: &Poseidon2WideCols<AB::Var> = (*cols).borrow();
+        let cols = if use_sbox_3 {
```

Comment:
same comment here, but again, compiler is probably making this ifs go away anyway